File: Task-03/Python/.py
def sieve_of_eratosthenes(n):
    is_prime = [True] * (n + 1)
    is_prime[0] = is_prime[1] = False

    for current_num in range(2, int(n ** 0.5) + 1):
        if is_prime[current_num]:
            for multiple in range(current_num * current_num, n + 1, current_num):
                is_prime[multiple] = False

    prime_numbers = [num for num in range(2, n + 1) if is_prime[num]]
    return prime_numbers

try:
    n = int(input("Enter a number (n): "))
    if n < 2:
        print("Prime numbers start from 2.")
    else:
        prime_numbers = sieve_of_eratosthenes(n)
        print(f"Prime numbers up to {n}:")
        print(prime_numbers)
except ValueError:
    print("Please enter a valid integer.")

File: Task-03/Rust/.rs
use std::io;

// Function to check if a number is prime
fn is_prime(num: u32) -> bool {
    if num <= 1 {
        return false;
    }
    for i in 2..=num / 2 {
        if num % i == 0 {
            return false;
        }
    }
    true
}

fn main() {
    println!("Enter a number (n):");

    let mut n = String::new();
    io::stdin()
        .read_line(&mut n)
        .expect("Failed to read line");

    let n: u32 = match n.trim().parse() {
        Ok(num) => num,
        Err(_) => {
            println!("Invalid input. Please enter a valid number.");
            return;
        }
    };

    if n < 2 {
        println!("Prime numbers start from 2.");
    } else {
        println!("Prime numbers up to {}:", n);
        for i in 2..=n {
            if is_prime(i) {
                println!("{}", i);
            }
        }
    }
}
